File: app/src/main/java/com/example/myfirstapp/MainActivity.java
package com.example.myfirstapp;

import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    public static final String EXTRA_MESSAGE = "com.example.myfirstapp.MESSAGE";
    public static final String SHARED_PREFS = "sharedPrefs";
    public static final String NAME = "name";
    private String name="";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        loadData();
        updateViews();
    }

    /** Called when the user taps the Send button */
    public void sendMessage(View view) {
        Intent intent = new Intent(this, DisplayMessageActivity.class);
        EditText editText = (EditText) findViewById(R.id.editText);
        String message = editText.getText().toString();
        //String [] send = new String[2];
        //send[0] = message;
        //intent.putExtra("list",send);
        intent.putExtra(EXTRA_MESSAGE, message);
        startActivity(intent);
        saveData();
    }

    public void saveData() {
        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        EditText editText = (EditText) findViewById(R.id.editText);
        editor.putString(NAME, editText.getText().toString());
        editor.apply();
        Toast.makeText(this, "Data saved", Toast.LENGTH_SHORT).show();
    }
    public void loadData() {
        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        name = sharedPreferences.getString(NAME, "");
    }
    public void updateViews() {
        EditText editText = (EditText) findViewById(R.id.editText);
        editText.setText(name);
    }
}

File: app/src/main/java/com/example/myfirstapp/MainActivity.kt
package com.example.myfirstapp

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var name: String? = ""
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        loadData()
        updateViews()
    }

    /** Called when the user taps the Send button  */
    fun sendMessage(view: View?) {
        val intent = Intent(this, DisplayMessageActivity::class.java)
        val editText = findViewById<View>(R.id.editText) as EditText
        val message = editText.text.toString()
        //String [] send = new String[2];
        //send[0] = message;
        //intent.putExtra("list",send);
        intent.putExtra(EXTRA_MESSAGE, message)
        startActivity(intent)
        saveData()
    }

    fun saveData() {
        val sharedPreferences = getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        val editText = findViewById<View>(R.id.editText) as EditText
        editor.putString(NAME, editText.text.toString())
        editor.apply()
        Toast.makeText(this, "Data saved", Toast.LENGTH_SHORT).show()
    }

    fun loadData() {
        val sharedPreferences = getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        name = sharedPreferences.getString(NAME, "")
    }

    fun updateViews() {
        val editText = findViewById<View>(R.id.editText) as EditText
        editText.setText(name)
    }

    companion object {
        const val EXTRA_MESSAGE = "com.example.myfirstapp.MESSAGE"
        const val SHARED_PREFS = "sharedPrefs"
        const val NAME = "name"
    }
}